homeassistant/components/huisbaasje/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for Huisbaasje integration."""
+import logging
+
+from huisbaasje import Huisbaasje, HuisbaasjeConnectionException, HuisbaasjeException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class HuisbaasjeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Huisbaasje."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        errors = {}
+
+        try:
+            user_id = await self._validate_input(user_input)
+
+            _LOGGER.info("Input for Huisbaasje is valid!")
+
+            # Set user id as unique id
+            existing_config_entry = await self.async_set_unique_id(user_id)
+
+            # Abort if entry already exists
+            if existing_config_entry is not None:
+                return self.async_abort(reason="already_configured")
```

Comment:
These lines can be simplified by using another helper-function:
```
                await self.async_set_unique_id(user_id, raise_on_progress=False)
                self._abort_if_unique_id_configured()
```
I'm not sure whether the `raise_on_progress=False` is needed here.

Reply:
`raise_on_progress` should be set to False, because the user just went through all the trouble of passing user/pass. The old one will be replaced with this new one.

Reply:
I'm sorry, but I just converted it to this because somebody else told me the solution you're suggesting uses a private method which should not be used, which sounds more reasonable to me. So if you're okay with that, i will leave this as is.

Reply:
`_abort_if_configured` is not a private method. It's a protected method.